pkg/policyreport/reportcontroller.go

```diff
@@ -614,18 +619,34 @@ func mergeRequests(ns *v1.Namespace, requestsGeneral interface{}) (*unstructured
 		}
 
 		req := &unstructured.Unstructured{Object: obj}
-		setReport(req, ns)
+		setReport(req, ns, kyverno)
 
 		return req, aggregatedRequests, nil
 	}
 
 	return nil, nil, nil
 }
 
-func setReport(reportUnstructured *unstructured.Unstructured, ns *v1.Namespace) {
+func setReport(reportUnstructured *unstructured.Unstructured, ns, kyverno *v1.Namespace) {
 	reportUnstructured.SetAPIVersion(report.SchemeGroupVersion.String())
 	reportUnstructured.SetLabels(LabelSelector.MatchLabels)
 
+	if kyverno != nil {
+		controllerFlag := true
+		blockOwnerDeletionFlag := true
+
+		reportUnstructured.SetOwnerReferences([]metav1.OwnerReference{
+			{
+				APIVersion:         "v1",
+				Kind:               "Namespace",
+				Name:               kyverno.GetName(),
+				UID:                kyverno.GetUID(),
+				Controller:         &controllerFlag,
+				BlockOwnerDeletion: &blockOwnerDeletionFlag,
```

Comment:
Do we want to block deletion of the Kyverno namespace till all policy reports are deleted? What is the behavior if we do not?

Reply:
I guess you meant "3. Test the deletion of the policy reports when the Kyverno namespace is **deleted** for the normal/success paths" ?

Reply:
Removed `BlockOwnerDeletion` and tested the namespace deletion, both cpolr and polr were removed.
```
✗ kg cpolr -o yaml | grep owner -A7
    ownerReferences:
    - apiVersion: v1
      controller: true
      kind: Namespace
      name: kyverno
      uid: 294d6204-dec6-4c6b-a994-bc382d9f719a
    resourceVersion: "817446"
```